utils/parse_config.py

```diff
@@ -31,18 +36,18 @@ def generate_gear_args(gear_context):
         log.info("No input provided. Will use files of type DICOM from session.")
 
     if download is False:
-        gear_args["infile"] = infile
-        gear_args["parent_acq"] = gear_context.get_input("file")["hierarchy"].get("id")
-        gear_args["session_id"] = gear_context.destination["id"]
+        gear_kwargs["infile"] = infile
+        gear_kwargs["parent_acq"] = gear_context.get_input("file")["hierarchy"].get("id")
+        gear_kwargs['session_id'] = fw.get_acquisition(gear_kwargs["parent_acq"]).parents.session
 
     else:
         # Alternatively, if no input is provided, all DICOM files in the session are
         # downloaded and used as input
-        gear_args["session_id"] = gear_context.destination["id"]
-        gear_args["api_key"] = gear_context.get_input("api_key")["key"]
-        gear_args["input_dir"] = "/flywheel/v0/input"
-
-    gear_args_formatted = pprint.pformat(gear_args)
+        gear_kwargs['session_id'] = gear_context.destination["id"]
+        gear_kwargs['input_dir'] = "/flywheel/v0/input"
+    
+    
+    gear_args_formatted = pprint.pformat(gear_kwargs)
```

Comment:
We need to pop out the API key so it doesn't print to the log file. That's my bad for not catching it before.